pkg/addon/addon.go

```diff
@@ -102,12 +102,12 @@ func GetAddonsFromReader(r AsyncReader, opt ListOptions) ([]*types.Addon, error)
 	if err != nil {
 		return nil, err
 	}
+	var l sync.Mutex
 	for _, subItem := range items {
 		if subItem.GetType() != "dir" {
 			continue
 		}
 		wg.Add(1)
-		var l sync.Mutex
 		go func(item Item) {
 			defer wg.Done()
```

Comment:
the `errChan` is going to block the whole function if error occurs more than twice:

![image](https://user-images.githubusercontent.com/2173670/144818065-f10dab66-d718-4539-be21-f42f5cf54704.png)

Reply:
fixed